test/stress_tests/test_actor_event_loop.py

```diff
@@ -0,0 +1,119 @@
+import ray
+import numpy as np
+import time
+from collections import Counter
+import json
+import sys
+
+LOCAL = len(sys.argv) > 1
+
+# Start the Ray processes.
+if LOCAL:
+    print("Starting test locally...")
+    from ray.test.cluster_utils import Cluster
+    cluster = Cluster(
+        initialize_head=True,
+        head_node_args={
+            "num_cpus": 10,
+            "_internal_config": json.dumps({
+                "initial_reconstruction_timeout_milliseconds": 1000,
+                "num_heartbeats_timeout": 10,
+            })
+        })
+    num_nodes = 10
+    for i in range(num_nodes - 1):
+        node_to_kill = cluster.add_node(
+            num_cpus=10,
+            num_gpus=10,
+            _internal_config=json.dumps({
+                "initial_reconstruction_timeout_milliseconds": 1000,
+                "num_heartbeats_timeout": 10,
+            }))
+    ray.init(redis_address=cluster.redis_address)
+else:
+    num_nodes = 100
+    ray.init(redis_address="localhost:6379")
+
+
+@ray.remote(num_gpus=1)
+class EventLoop(object):
+    def __init__(self, num_children):
+        self.children = [Child.remote() for _ in range(num_children)]
+        self.nodes = ray.get([child.node.remote() for child in self.children])
+        self.counts = Counter()
+
+        self.tasks = [child.ping.remote() for child in self.children]
+        self.index = {task: i for i, task in enumerate(self.tasks)}
+
+    def ping(self):
+        ready, unready = ray.wait(self.tasks, num_returns=1)
+        for task in ready:
+            i = self.index[task]
+            new_task = self.children[i].ping.remote()
+            self.tasks[i] = new_task
+            self.index[new_task] = i
+            self.counts[self.nodes[i]] += 1
+
+    def counts(self):
+        return dict(self.counts)
```

Comment:
Did you encounter https://github.com/ray-project/ray/issues/3846? I fixed it in Arrow, so it should be fixed as soon as we upgrade Arrow.